docs/quickstart/cpp.md

```diff
@@ -16,34 +16,29 @@ working example.</p>
 
 #### Install gRPC
 
-To install gRPC on your system, follow the instructions to build from source
-[here](https://github.com/grpc/grpc/blob/master/INSTALL.md).
+To install gRPC on your system, follow the [instructions to build from source]
+(https://github.com/grpc/grpc/blob/master/INSTALL.md).
+
+To run the example code, please ensure `pkg-config` is installed on your
+machine before you build and install gRPC in the previous step, since the
+example `Makefile`s try to look up the installed gRPC path using `pkg-config`.
+On Debian-based systems like Ubuntu, this can usually be done via
+`sudo apt-get install pkg-config`.
 
 #### Install Protocol Buffers v3
 
 While not mandatory, gRPC usually leverages Protocol Buffers v3 for service
-definitions and data serialization. If you don't already have it installed on
+definitions and data serialization, and our example code uses Protocol
+Buffers as well as gRPC. If you don't already have it installed on
 your system, you can install the version cloned alongside gRPC. First ensure
-that you are running these commands from within your cloned gRPC repository
+that you are running these commands in the gRPC tree you just built in the
 from the previous step.
 
 ```sh
-$ git submodule update --init
-$ cd grpc/third_party/protobuf
-$ ./autogen.sh
-$ ./configure
-$ make
-$ sudo make install
+$ cd third_party/protobuf
+$ make install
```

Comment:
I have mixed feelings about this.
I thinks simplifying the docs is useful, but on the other hand, I'm not sure if installing protobuf from grpc's submodule is something worth recommending (it's probably fine for the purposes of the examples)

As a side note, I think calling "make install" is actually an anti-pattern IMHO (there is no way to uninstall and that has bitten many users in the past). I'd be much happier if our example instructions mentioned that "make install" is something you shouldn't do and what are the risks.

Reply:
The issue is that installing protobuf from the tar.gz file won't register it on pkg-config and will cause our examples to fail